File: src/Samples/SharpPie/src/SharPie/PropertyTemplates/InputElementPropertyTemplate.php
<?php

namespace SharPie\PropertyTemplates;

use Peachpie\Avalonia\ControlsTemplates\PropertyTemplates\InputElementPropertyTemplate;
use SharPie\Input\UxInputElement;

trait UxInputElementPropertyTemplate
{
    /**
     * @param bool $value
     * @return UxInputElement
     */
    public function Focusable(bool $value) : UxInputElement
    {
        return InputElementPropertyTemplate::Focusable($this, $value);
    }

    /**
     * @param bool $value
     * @return UxInputElement
     */
    public function IsEnabled(bool $value) : UxInputElement
    {
        return InputElementPropertyTemplate::IsEnabled($this, $value);
    }

    /*
    public function Cursor(Cursor $value) : Object
    {
        return InputElementPropertyTemplate::Cursor($this, $value);
    }
    */

    /**
     * @param bool $value
     * @return UxInputElement
     */
    public function IsHitTestVisible(bool $value) : UxInputElement
    {
        return InputElementPropertyTemplate::IsHitTestVisible($this, $value);
    }

    /**
     * @param bool $value
     * @return UxInputElement
     */
    public function IsTabStop(bool $value) : UxInputElement
    {
        return InputElementPropertyTemplate::IsTabStop($this, $value);
    }

    /**
     * @param int $value
     * @return UxInputElement
     */
    public function TabIndex(int $value) : UxInputElement
    {
        return InputElementPropertyTemplate::TabIndex($this, $value);
    }


}

File: src/Peachpie.Avalonia/ControlsTemplates/PropertyTemplates/InputElementPropertyTemplate.cs
using System.Runtime.CompilerServices;
using Avalonia.Data;
using Avalonia.Data.Converters;
using Avalonia.Input;
using Pchp.Core;

namespace Peachpie.Avalonia.ControlsTemplates.PropertyTemplates;

public static class InputElementPropertyTemplate
{
   
   public static PhpValue Focusable(InputElement control, bool value = default, BindingMode? bindingMode = null,
      IValueConverter converter = null, object bindingSource = null,
      [CallerArgumentExpression("value")] string ps = null)
   {
      return PhpValue.FromClass(control._setEx(InputElement.FocusableProperty, ps, () => control.Focusable = value, bindingMode,
         converter, bindingSource));
   }
   

   public static PhpValue IsEnabled(InputElement control, bool value = default, BindingMode? bindingMode = null,
      IValueConverter converter = null, object bindingSource = null,
      [CallerArgumentExpression("value")] string ps = null)
   {
      return PhpValue.FromClass(control._setEx(InputElement.IsEnabledProperty, ps, () => control.IsEnabled = value, bindingMode, converter,
         bindingSource));
   }
   
   
   public static PhpValue Cursor(InputElement control, Cursor value = default, BindingMode? bindingMode = null,
      IValueConverter converter = null, object bindingSource = null,
      [CallerArgumentExpression("value")] string ps = null)
   {
      return PhpValue.FromClass(control._setEx(InputElement.CursorProperty, ps, () => control.Cursor = value, bindingMode, converter,
         bindingSource));
   }
   
   public static PhpValue IsHitTestVisible(InputElement control, bool value = default, BindingMode? bindingMode = null,
      IValueConverter converter = null, object bindingSource = null,
      [CallerArgumentExpression("value")] string ps = null)
   {
      return PhpValue.FromClass(control._setEx(InputElement.IsHitTestVisibleProperty, ps, () => control.IsHitTestVisible = value,
         bindingMode,
         converter, bindingSource));
   }
   
   public static PhpValue IsTabStop(InputElement  control, bool value = default, BindingMode? bindingMode = null,
      IValueConverter converter = null, object bindingSource = null,
      [CallerArgumentExpression("value")] string ps = null)
   {
      return PhpValue.FromClass(control._setEx(InputElement.IsTabStopProperty, ps, () => control.IsTabStop = value, bindingMode, converter,
         bindingSource));
   }
   
   public static PhpValue TabIndex(InputElement control, int value = default, BindingMode? bindingMode = null,
      IValueConverter converter = null, object bindingSource = null,
      [CallerArgumentExpression("value")] string ps = null)
   {
      return PhpValue.FromClass(control._setEx(InputElement.TabIndexProperty, ps, () => control.TabIndex = value, bindingMode, converter,
         bindingSource));
   }

}